File: drive_java/src/main/java/com/github/ximik3/gradle/PluginType.java
package com.github.ximik3.gradle;

enum PluginType {
    GAM(1),
    MIMI(2),
    UNDEFINED(0);

    int value;

    PluginType(int value) {
        this.value = value;
    }

    public static PluginType parse(int value) {
        switch (value) {
            case 1:
                return GAM;
            case 2:
                return MIMI;
            default:
                return UNDEFINED;
        }
    }
}

File: drive_kotlin/src/main/kotlin/com/github/ximik3/gradle/enums/PluginType.kt
package com.github.ximik3.gradle.enums

enum class PluginType(var value: Int) {
    GAM(1), MIMI(2), UNDEFINED(0);

    companion object {
        fun parse(value: Int): PluginType {
            return when (value) {
                1 -> GAM
                2 -> MIMI
                else -> UNDEFINED
            }
        }
    }
}